pkg/netstore/netstore.go

```diff
@@ -57,31 +70,70 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 				go s.recoveryCallback(addr, targets)
 				return nil, ErrRecoveryAttempt
 			}
-			stamp, err := ch.Stamp().MarshalBinary()
-			if err != nil {
-				return nil, err
-			}
-
-			putMode := storage.ModePutRequest
-			if mode == storage.ModeGetRequestPin {
-				putMode = storage.ModePutRequestPin
-			}
-
-			cch, err := s.validStamp(ch, stamp)
-			if err != nil {
-				// if a chunk with an invalid postage stamp was received
-				// we force it into the cache.
-				putMode = storage.ModePutRequestCache
-				cch = ch
-			}
-
-			_, err = s.Storer.Put(ctx, putMode, cch)
-			if err != nil {
-				return nil, fmt.Errorf("netstore retrieve put: %w", err)
-			}
+			s.put(ch, mode)
```

Comment:
nitpicking, but writing this as:
```go
s.wg.Add(1)
go s.put(ch, mode)
```
makes it easier to understand that there's something asynchronous which is happening here without having to navigate to the method definition.